### DISCUSSION TITLE
The query doesn't return full properties after creating

### DISCUSSION THREAD

**Discussion by Jedliu** [mikro-orm/mikro-orm#3186] [Q&A] [answered]
**I'm not sure if this was designed intentionally or a bug. Thanks,**

**Describe the bug**
I have a Book entity and the description property is optional. After first creating, the query result doesn't contain a description property even via a findOne call. It'll return the result with the description to be null via an API call again.

**Stack trace**
Return the created entity or via findOne function
```
async createBook(dto: CreateBookDTO): Promise<MikroBook> {
    const { title, author } = dto;
    const book = new MikroBook(title, author);

    await this.bookRepository.persistAndFlush(book);
    return this.findOne(book.id);
  }
```
No description property in the result.
```
{
    "id": 5,
    "title": "Hello world",
    "author": "Bill"
}
```
When try to findOne from an API call again, you will see a null description.
```
{
    "id": 5,
    "title": "Hello world",
    "author": "Bill",
    "description": null
}
```

**Expected behavior**
Should return a null property same as the via findOne function.

**Versions**
| mikro-orm | 5.1.5 |
| your-driver | postgresql |

**Comment by B4nan** [comment, marked as answer]
Yes, this is intentional, if you want to have the `null` always there, add it to the property initializer:

```ts
// note we need to provide the type explicitly as reflect-metadata cant extract information from union types
@Property({ type: 'string' })
description: string | null = null;
```

The difference is that with SQL drivers all properties returned from database will have explicit `null`, unlike entities you create via constructor (their values are not coming from db). Alternative to explicit `null` default values is `forceUndefined` config option which will make those `null`s coming from database be hydrated as `undefined`, so your second API call would also _not_ have that description prop. Maybe we should switch the default in next major.

**Reply by Jedliu** [reply]
This was explained very clearly. I prefer to have the same data structure so I will set it to default null. Thank you for your prompt reply as always!
